modules/plugin/jdbc/jdbc-oracle/src/test/java/org/geotools/data/oracle/OracleDataStoreOnlineTest.java

```diff
@@ -77,5 +78,27 @@ public void testCreateSpatialIndexNameTooLong() throws Exception {
         // used to fail here
         dataStore.createSchema(featureType);
     }
+    
+    public void testCreateLongVarChar() throws Exception {
+       SimpleFeatureTypeBuilder builder = new SimpleFeatureTypeBuilder();
+        builder.setName(tname("ft2"));
+        builder.setNamespaceURI(dataStore.getNamespaceURI());
+        builder.setCRS(DefaultGeographicCRS.WGS84);
+        builder.add(aname("geometry_one_two_three_four"), Geometry.class);
+        builder.add(aname("longvar"), String.class);
+        
+
+        SimpleFeatureType featureType = builder.buildFeatureType();
+        
+        dataStore.createSchema(featureType);
+        SimpleFeatureBuilder fBuilder = new SimpleFeatureBuilder(featureType);
+        fBuilder.add(null);
+        StringBuffer vBuffer = new StringBuffer(4000);
+        //to be honest I can't tell from the oracle docs if 4000 or 3999 is the actual limit
+        for(int i=0;i<3999;i++) {
+        	vBuffer.append("x");
+        }
+        fBuilder.add(vBuffer.toString());
```

Comment:
Unsure what is happening here... is the feature builder going to throw an exception if we go above the limits?
Btw, there is also FeatureTypes.getFieldLength that could come in handy for a test